srtcore/core.cpp

```diff
@@ -10033,25 +10115,56 @@ int32_t CUDT::bake(const sockaddr_any& addr, int32_t current_cookie, int correct
     }
 }
 
-// XXX This is quite a mystery, why this function has a return value
-// and what the purpose for it was. There's just one call of this
-// function in the whole code and in that call the return value is
-// ignored. Actually this call happens in the CRcvQueue::worker thread,
-// where it makes a response for incoming UDP packet that might be
-// a connection request. Should any error occur in this process, there
-// is no way to "report error" that happened here. Basing on that
-// these values in original UDT code were quite like the values
-// for m_iReqType, they have been changed to URQ_* symbols, which
-// may mean that the intent for the return value was to send this
-// value back as a control packet back to the connector.
-//
+// This function does the same as the fragment that prepares the handshake
+// in CUDT::createSrtHandshake. This should only copy one type of extension
+// with contents possible to be specified as a string. Many extensions have
+// specific ways of extracting the data into the extension, hence this is
+// not refactored into single functions.
+size_t CUDT::addHandshakeExtension(char* data, int cmd, size_t hs_size, std::string contents)
+{
+    size_t ra_size = hs_size / sizeof(int32_t);
+
+    // Now attach the SRT handshake for HSREQ
+    size_t    offset = ra_size;
+    uint32_t *p      = reinterpret_cast<uint32_t *>(data);
+    // NOTE: since this point, ra_size has a size in int32_t elements, NOT BYTES.
+
+    // The first 4-byte item is the CMD/LENGTH spec.
+    uint32_t *pcmdspec = p + offset; // Remember the location to be filled later, when we know the length
+    ++offset;
+
+    // Now prepare the string with 4-byte alignment. The string size is limited
+    // to half the payload size. Just a sanity check to not pack too much into
+    // the conclusion packet.
+    size_t size_limit = m_iMaxSRTPayloadSize / 2;
+
+    if (contents.size() >= size_limit)
+    {
+        contents = "#!::="; // size error
```

Comment:
Just came to my mind: probably this isn't possible here, as the size limit should be checked when setting streamid.